### PR TITLE
Compact and defrag before backup

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/19844

This PR runs a compact and defrag on the etcd data store before taking the backup.

### CODE REVIEW

File: pkg/etcd/etcdv3.go [hunk 1]
```diff
@@ -144,3 +136,117 @@ func (b V3Backup) getTmpDir() string {
...
+
+	type endpointStatusOutputStatus struct {
+		Header endpointStatusOutputStatusHeader
+	}
+
+	type endpointStatusOutput struct {
+		Status endpointStatusOutputStatus
+	}
+
+	// [
```
Comment (AverageMarcus) [inline pkg/etcd/etcdv3.go:222]: ???

Comment (whites11) [reply pkg/etcd/etcdv3.go:222]: that is the JSON I am about to parse after the comment

Comment (AverageMarcus) [reply pkg/etcd/etcdv3.go:222]: Do we need the comment left there? There's also an example of it in the test if needed.

Comment (whites11) [reply pkg/etcd/etcdv3.go:222]: okay dropping

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**